testsuite/utils.py

```diff
@@ -124,6 +124,11 @@ def extract_response(response, header="Simple", key="data"):
     :param response: http response
     :return: Extracted value
     """
+
+    # Returning None for non-200 responses because the content of such responses might not be a valid JSON
+    if response.status_code != 200:
```

Comment:
I would probably rather have a more generic fix if we ever decide to change how the response is done, WDYT?

```
data = weakget(response.json())["headers"][header]
return weakget(json.loads(data % "{}")[key]
```

Reply:
The issue with this is, that `response.json()` throws an error when the body of the response is an empty string

Reply:
@pehala  @azgabur  If you still have a mental power to discuss this, here are two alternatives I considered:
```
    try:
        response.json()
    except JSONDecodeError:
        return weakget(None)
```
```
    if len(response.content) == 0:
        return weakget(None)
```
The former is not optimal imo because there is a potential to "swallow" a valid error - if there is an invalid json there for whatever reason - I can't think of any such reason but still.

The latter is actually quite ok as well. If there is nothing there just return None and try to do the parsing otherwise.  In case of invalid json this would throw an error but I think that it is a good thing.

If you strongly prefer one of these alternatives just let me know and I can create another PR